packages/reports/addon/helpers/report-action.js

```diff
@@ -1,26 +1,29 @@
 /**
- * Copyright 2017, Yahoo Holdings Inc.
+ * Copyright 2018, Yahoo Holdings Inc.
  * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
  */
-import Ember from 'ember';
-import RouteAction from 'ember-route-action-helper/helpers/route-action';
+import Helper from '@ember/component/helper';
+import { inject as service } from '@ember/service';
+import { get } from '@ember/object';
+import { assert } from '@ember/debug';
 import { ReportActions } from 'navi-reports/services/report-action-dispatcher';
 
-const ROUTE_ACTION = 'onUpdateReport';
+export default Helper.extend({
+  /**
+   * @property {Service} reportActionDispatcher
+   */
+  reportActionDispatcher: service(),
 
-export default RouteAction.extend({
   /**
-   * Returns a closure action that validates the report action
-   * name and sends it out to be dispatched by the report-action-dispatcher
+   * Validates a report action and then dispatches it via the report action dispatcher
    *
-   * @method compute
-   * @override
-   * @params {Array} - The positional arguments to the helper
-   * @returns {Function} - Closure action
+   * @method reportAction
+   * @param {Array} array with the name of the action to dispatch
+   * @returns {Function} Closure action
    */
-  compute([reportAction, ...params]) {
-    let actionName = ReportActions[reportAction];
-    Ember.assert(`The action name "${actionName}" is not a valid report action`, actionName);
-    return this._super([ROUTE_ACTION, actionName, ...params]);
+  compute([action, ...params]) {
```

Comment:
When making an action helper, make sure to:
- execute in the runloop: 
https://github.com/DockYard/ember-route-action-helper/blob/v2.0.6/addon/helpers/route-action.js#L50

- make function a "real" action 
https://github.com/DockYard/ember-route-action-helper/blob/v2.0.6/addon/helpers/route-action.js#L53